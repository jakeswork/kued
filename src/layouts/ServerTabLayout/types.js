import {
  shape,
  node,
  arrayOf,
  string,
  func,
} from 'prop-types';

const dropdownOptionType = shape({
  value: string,
  label: string,
});
const dropdownOptionDefault = {
  value: '',
  label: '',
};

export const propTypes = {
  classes: shape({}).isRequired,
  children: node,
  dropdownOptions: arrayOf(dropdownOptionType),
  dropdownOptionDefault: dropdownOptionType,
  dropdownOnChange: func,
};

export const defaultProps = {
  children: null,
  dropdownOptions: [dropdownOptionDefault],
  dropdownOptionDefault,
  dropdownOnChange: () => {},
};
