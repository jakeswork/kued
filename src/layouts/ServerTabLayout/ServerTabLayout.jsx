import React from 'react';
import { TabPanel } from 'react-tabs';

import { propTypes, defaultProps } from './types';

const ServerTabLayout = ({ classes, children, ...props }) => (
  <TabPanel {...props}>
    <div className={classes.root}>
      { children }
    </div>
  </TabPanel>
);

ServerTabLayout.propTypes = propTypes;

ServerTabLayout.defaultProps = defaultProps;

ServerTabLayout.tabsRole = 'TabPanel';

export default ServerTabLayout;
