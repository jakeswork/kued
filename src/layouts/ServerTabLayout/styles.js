export default {
  root: {
    padding: '32px 0',
    width: '100%',
    textAlign: 'center',
  },
};
