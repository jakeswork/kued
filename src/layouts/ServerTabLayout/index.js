import withStyles from 'react-jss';

import ServerTabLayout from './ServerTabLayout';
import styles from './styles';

export default withStyles(styles)(ServerTabLayout);
