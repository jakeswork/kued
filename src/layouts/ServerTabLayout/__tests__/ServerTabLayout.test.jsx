import React from 'react';
import { shallow } from 'enzyme';

import ServerTabLayout from '../ServerTabLayout';

const defaultProps = {
  classes: {},
};

describe('The ServerTabLayout instance', () => {
  describe('When called with default props', () => {
    it('should render without throwing an error', () => {
      const wrapper = shallow(<ServerTabLayout {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
