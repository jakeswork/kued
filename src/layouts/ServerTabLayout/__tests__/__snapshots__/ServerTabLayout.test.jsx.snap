// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ServerTabLayout instance When called with default props should render without throwing an error 1`] = `
<TabPanel
  className="react-tabs__tab-panel"
  forceRender={false}
  selectedClassName="react-tabs__tab-panel--selected"
>
  <div />
</TabPanel>
`;
